packages/juggler/src/legacy-juggler-bridge.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/juggler
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+export const jugglerModule = require('loopback-datasource-juggler');
+
+import {MixinBuilder} from './mixin';
+import {Class, ObjectType, Options, AnyType} from './common';
+
+import {juggler} from './loopback-datasource-juggler';
+
+export * from './loopback-datasource-juggler';
+
+/* tslint:disable-next-line:variable-name */
+export const DataSource = jugglerModule.DataSource as typeof juggler.DataSource;
+/* tslint:disable-next-line:variable-name */
+export const ModelBase = jugglerModule.ModelBaseClass as typeof juggler.ModelBase;
+
+/**
+ * This is a bridge to the legacy DAO class. The function mixes DAO methods
+ * into a model class and attach it to a given data source
+ * @param modelClass {} Model class
+ * @param ds {DataSource} Data source
+ * @returns {} The new model class with DAO (CRUD) operations
+ */
+export function bindModel<T extends typeof juggler.ModelBase>(modelClass: T,
+  ds: juggler.DataSource): T {
+  const boundModelClass = class extends modelClass {};
+  boundModelClass.attachTo(ds);
+  return boundModelClass;
+}
+
+import {Entity} from './model';
+import {Filter, Where} from './query';
+import {EntityCrudRepository} from './repository';
+
+function isPromise<T>(p: AnyType): p is Promise<T> {
+  return p !== null && typeof p === 'object' && typeof p.then === 'function';
+}
+
+function getPromise<T>(p: juggler.PromiseOrVoid<T>): Promise<T> {
+  if (isPromise(p)) {
+    return p;
+  } else {
+    return Promise.reject(new Error('The value should be a Promise: ' + p));
+  }
+}
+
+export class DefaultCrudRepository<T extends Entity, ID>
+implements EntityCrudRepository<T, ID> {
+  modelClass: typeof juggler.PersistedModel;
+
+  constructor( modelClass: typeof juggler.PersistedModel, dataSource: juggler.DataSource) {
+    this.modelClass = bindModel(modelClass, dataSource);
+  }
+
+  create(entity: ObjectType<T>, options?: Options): Promise<T> {
+    return getPromise(this.modelClass.create(entity, options));
+  }
+
+  createAll(entities: ObjectType<T>[], options?: Options): Promise<T[]> {
+    return getPromise(this.modelClass.create(entities, options));
+  }
+
+  save(entity: ObjectType<T>, options?: Options): Promise<T> {
+    const idName = this.modelClass.definition.idName();
+    let id;
+    if (typeof entity.getId === 'function') {
+      id = entity.getId();
+    } else {
+      id = entity[idName];
+    }
+    if (id == null) {
+      return this.create(entity, options);
+    } else {
+      return this.replaceById(id, entity, options).
```

Comment:
This implementation does not allow the following usage:

```js
// id "pen1" does not exist yet
const inst = new Product({id: 'pen1', name: 'Pen'});
inst.save();
```

I don't mind must, just pointing out an edge case. If it was an intentional decision then you can ignore my comment. If it was not intentional and you want to land this PR soon, then I am fine with changing this later.